src/overridelogger.jl

```diff
@@ -0,0 +1,22 @@
+"""
+    LevelOverrideLogger(level, logger)
+
+A logger that allows overriding the log level of a child level.
+Useful in debugging scenarios, where it is desirable to ignore
+the log level any other logger may have set.
+"""
+struct LevelOverrideLogger{T <: AbstractLogger} <: AbstractLogger
+    level::LogLevel
+    logger::T
+end
+
+
+handle_message(logger::LevelOverrideLogger, args...; kwargs...) =
+    handle_message(logger.logger, args...; kwargs...)
+
+function shouldlog(logger::LevelOverrideLogger, level, args...)
+    level >= logger.level && shouldlog(logger.logger, level, args...)
```

Comment:
The 1st way it overrides is by only returning its own `level` from `min_enabled_level` (other loggers typically return the `max` of itself and child loggers). Also in this case, it's only checking its own level and then dispatching to it's nested logger `shouldlog`; as far as I understand, loggers don't typically check levels in `shouldlog` because it happens before w/ `min_enabled_level`. 

We're also just overriding the _level_; if there are other filters from the nested logger, then those will still be respected. If the nested logger _does_ check levels in their `shouldlog`, then those will be respected. It's a bit tricky because it's hard to know exactly what the nested logger is going to do in `shouldlog`, but from initial tests, this seems to be working as expected.

Reply:
It looks like the `comp_shouldlog` which is called by all the composable loggers in the package checks both `level` and `shouldlog`.
https://github.com/JuliaLogging/LoggingExtras.jl/blob/396afbb317c400e3b04a51f5dfd3e1880e087653/src/LoggingExtras.jl#L35

There is a comment above it saying that because the loggering system itself won't check the `level` of child loggers.

I think I ran into a bug before I had this with something like `TeeLogger(MinLevelFilter(Debug, ...), MinLevelFilter(Warn, ...))`
Which should have the `TeeLogger` saying it's minlevel is `Debug` since it needs to recieve Debug level messages.
But then those need only to be accepted by one of the loggers and not by the other.

I think this can possibly be fixed by moving the logic from the recieving logger, up into the sending logger.
Which would better match how the logging system works outside the compositional structure.
So it only forwards to children that say they can accept it.
(vs right now where the child decides that it won't log it because of it's level)

Reply:
Can a test be added to check this case please

Reply:
This `shouldlog` method is exercised in the tests here: https://github.com/JuliaLogging/LoggingExtras.jl/blob/73f17b8ace3d13db00a7e925483a1d9776c7eb7a/test/runtests.jl#L254. i.e. we're ensuring that with the LevelOverrideLogger, that the LevelOverride `Debug` level is "overriding" the TestLogger `Info` level.

Is there a more specific/different test desired here?

Reply:
yes,
I want to see:

```julia
LoggingExtras.with(level=Debug, verbosity=1) do 
    with_logger(MinLevelLogger(Info, test_logger)) do
        @debug "This should show up, even though it is behind 2 info level filters"
    end
end
```

Reply:
I'm not sure this can be made to work due to the way MinLevelLogger has it's `shouldlog` and `min_level_enabled` defined; i.e. the presence of a `MinLevelLogger` in a stack of loggers effectively "poisons" any other kind of level filtering. It's `min_enabled_level` calls `max` of its own level and its child logger, so it's going to raise the level of a LevelOverrideLogger. And because MinLevelLogger `shouldlog` calls `comp_shouldlog` of its child logger, and because `comp_shouldlog` checks the `min_enabled_level` of the child logger, it means LevelOverrideLogger gets overridden, even if we were to get back the min_enabled check.

It's a similar issue if LevelOverride wraps a MinLevelLogger; the LevelOverride will pass the level check, but then it calls the `shouldlog` of its child logger and the `MinLevelLogger` will "take over" as mentioned above.

I don't see an immediate/easy way to detangle all that without going big with an overhaul of all the other compositional loggers here, so I'm inclined to just put a note in the docs that the LevelOverrideLogger should be used independently from other level-filtering loggers.

Reply:
I think I know how to fix it.
Which is a big overhaul of all the other compositional loggers.
But I don't think it will take me long.
Lets put that test case in.
Mark it broken, and open an issue cross referencing it, then merge this.
Then later tonight or tomorrow I will fix it.

Reply:
Ok, added the broken test.